exe/Arguments.hs

```diff
@@ -1,37 +1,48 @@
 -- Copyright (c) 2019 The DAML Authors. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
-module Arguments(Arguments(..), getArguments) where
+module Arguments(Arguments, Arguments'(..), getArguments, IdeCmd(..)) where
 
 import Options.Applicative
+import HieDb.Run
 
+type Arguments = Arguments' IdeCmd
 
-data Arguments = Arguments
+data IdeCmd = Typecheck [FilePath] | DbCmd Command | LSP
+
+data Arguments' a = Arguments
     {argLSP :: Bool
```

Comment:
why is this field still needed?

Reply:
Backward compatibility - we don't want to break all existing `ghcide` command lines!

Reply:
But that can be handled in the parser, can't it?

Reply:
I couldn't get `optparse-applicative` to do this any other way.

Reply:
To do what, parse `--lsp`?

Reply:
Yeah

Reply:
I find that using `ApplicativeDo` to construct parsers clarifies things massively. Or just ue a smart constructor

Reply:
The field is still used in `Main.hs` though, isn't it?

Anyway, something like this should work:
```haskell
diff --git a/exe/Arguments.hs b/exe/Arguments.hs
index ddffd35..afa56e6 100644
--- a/exe/Arguments.hs
+++ b/exe/Arguments.hs
@@ -11,8 +11,7 @@ type Arguments = Arguments' IdeCmd
 data IdeCmd = Typecheck [FilePath] | DbCmd Command | LSP
 
 data Arguments' a = Arguments
-    {argLSP :: Bool
-    ,argsCwd :: Maybe FilePath
+    {argsCwd :: Maybe FilePath
     ,argsVersion :: Bool
     ,argsShakeProfiling :: Maybe FilePath
     ,argsTesting :: Bool
@@ -30,7 +29,7 @@ getArguments = execParser opts
 
 arguments :: Parser Arguments
 arguments = Arguments
-      <$> switch (long "lsp" <> help "Start talking to an LSP server")
+      <$ switch (long "lsp" <> help "Start talking to an LSP server" <> internal)
       <*> optional (strOption $ long "cwd" <> metavar "DIR" <> help "Change to this directory")
       <*> switch (long "version" <> help "Show ghcide and GHC versions")
       <*> optional (strOption $ long "shake-profiling" <> metavar "DIR" <> help "Dump profiling reports to this directory")
```
The `internal` is so that `--lsp` doesn't show up in help texts